pkg/cincinnati/cincinnati.go

```diff
@@ -51,14 +56,29 @@ func (err *Error) Error() string {
 	return fmt.Sprintf("%s: %s", err.Reason, err.Message)
 }
 
+// Run HTTP request in a goroutine and pass the response to f.
+func clientDo(ctx context.Context, req *http.Request, client http.Client, f func(*http.Response, error) error) error {
+	c := make(chan error, 1)
+	getUpdatesCtx, cancel := context.WithTimeout(ctx, getUpdatesTimeout)
+	defer cancel()
+	go func() { c <- f(client.Do(req.WithContext(getUpdatesCtx))) }()
+	select {
+	case <-getUpdatesCtx.Done():
```

Comment:
What's the story around this?  It seems to be "we don't trust `f` or `client.Do` or `req.WithContext` to respect `ctx`, so we'll wrap it in `clientDo` and also watch `getUpdatesCtx` ourselves" or some such.

Reply:
Not any of that. It seemed like a clean way to encapsulate waiting for either a timeout/cancel, an error or the response. But I'm not partial to it if you have something cleaner in mind.

Reply:
[This][1] seems sufficient to me.

[1]: https://github.com/openshift/cluster-version-operator/pull/83/files#diff-3a796940f9df2373810e52aae0d83795R62

Reply:
Removed clientDo method in commit [8ee0d52](https://github.com/openshift/cluster-version-operator/pull/410/commits/8ee0d52e67593922f965e8c48bc22275acc62bab)